### PR TITLE
Migrate usage of EKS to aws-sdk-go-v2

### LINKED ISSUE #4884: migrate to aws-sdk-go-v2 [8/9]
## Background

As part of #3215 we want to migrate to use the [aws-sdk-go v2](https://github.com/aws/aws-sdk-go-v2). This involves migrating all of the AWS API calls, roughly 500~ to use the new package. A spike was complete and its finding was documented [here](https://github.com/weaveworks/eksctl/issues/3215#issuecomment-1047099306) and that included a demo branch [here](https://github.com/weaveworks/eksctl/pull/4820/files). As the spike documents we can split out the work to migrate to v2 to be done in 9 parts, were we essentially use both v1 and v2 of the API while we migrate.

This ticket is to: remove usage of `v1` api for `EKS` and  replace it with `v2`. Consider doing in multiple PRs (where `EKS` and `EKSV2` can live along-side eachother) 

Blocked on #4877. Can be done in parallel to parts 2-8
Also, when migrating to a new call we will propagate the context from the command down to the AWS calls.

### PR DESCRIPTION
**Pull request by cPu1**
### Description

Migrates usage of EKS to aws-sdk-go-v2.

Closes #4884 

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/ctl/enable/flux.go [hunk 1]
```diff
@@ -50,7 +51,8 @@ func flux2Install(cmd *cmdutils.Cmd) error {
 	logger.Info("will install Flux v2 components on cluster %s", cmd.ClusterConfig.Metadata.Name)
 
 	if kubeconfAndContextNotSet(cmd.ClusterConfig.GitOps.Flux.Flags) {
-		ctl, err := cmd.NewProviderForExistingCluster()
+		ctx := context.TODO()
+		ctl, err := cmd.NewProviderForExistingCluster(ctx)
```
Comment (Himangini) [inline pkg/ctl/enable/flux.go:55]:
```suggestion
		ctl, err := cmd.NewProviderForExistingCluster(context.TODO())
```

**Review by Himangini — COMMENTED** [review]
small nits and questions about todos, otherwise everything looks good 👍🏻

**Review by Skarlso — APPROVED** [review]

File: pkg/ctl/utils/nodegroup_health.go [hunk 2]
```diff
@@ -60,18 +63,19 @@ func getNodeGroupHealth(cmd *cmdutils.Cmd, nodeGroupName string) error {
...
 	stackCollection := manager.NewStackCollection(ctl.Provider, cfg)
 	managedService := managed.NewService(ctl.Provider.EKS(), ctl.Provider.EC2(), stackCollection, cfg.Metadata.Name)
-	healthIssues, err := managedService.GetHealth(nodeGroupName)
+	healthIssues, err := managedService.GetHealth(ctx, nodeGroupName)
 	if err != nil {
 		return err
 	}
 
 	if len(healthIssues) == 0 {
+		// TODO
```
[the remark this answers is not in the document]
Comment (cPu1) [reply pkg/ctl/utils/nodegroup_health.go]: No, it's resolved.

File: pkg/fargate/fargate.go [hunk 3]
```diff
@@ -60,10 +61,7 @@ func toSelectors(in []*eks.FargateProfileSelector) []api.FargateProfileSelector
...
 //   AccessDeniedException: Account <account> is not authorized to use this service
 func IsUnauthorizedError(err error) bool {
-	awsErr, ok := errors.Cause(err).(awserr.Error)
-	if !ok {
-		return false
-	}
-
-	return awsErr.Code() == "AccessDeniedException"
+	var accessDeniedErr *ekstypes.AccessDeniedException
+	// TODO: test
```
[the remark this answers is not in the document]
Comment (cPu1) [reply pkg/fargate/fargate.go]: It was for testing it, it's done now.

**Pull request merged**